lib/opencensus_phoenix/instrumenter.ex

```diff
@@ -87,12 +90,43 @@ defmodule OpencensusPhoenix.Instrumenter do
     :ocp.with_span_ctx(parent_span_ctx)
   end
 
-  def span_name(conn) do
+  defp span_name(conn, _) do
+    controller_action(conn)
+  end
+
+  defp controller_action(conn) do
     controller = Phoenix.Controller.controller_module(conn)
     action = Phoenix.Controller.action_name(conn)
     "#{controller}.#{action}"
   end
 
+  # Version 1.4 or higher
+  phoenix_version_supports_route_info? =
+    Code.ensure_compiled?(Phoenix.Router) &&
+      :erlang.function_exported(Phoenix.Router, :route_info, 4)
+
+  if phoenix_version_supports_route_info? do
+    defp span_name(_conn, %{route: route}) do
+      route
+    end
+  else
+    defp span_name(conn, _) do
+      controller_action(conn)
+    end
+  end
+
+  if phoenix_version_supports_route_info? do
+    defp route_info(%{method: method, request_path: request_path, host: host} = conn) do
+      router = conn.private[:phoenix_router]
+
+      apply(Phoenix.Router, :route_info, [router, method, request_path, host])
```

Comment:
Why is this an apply instead of just `Phoenix.Router.route_info(router, method, request_path, host)`

Reply:
I guess it was because it was generating a compiler warning, but now it won't be compiled without that function available, so I'll make it a normal call.

Reply:
I haven't (and won't for a bit) have a chance to test these on a running phoenix application so if you have one on hand we should try it out before merging.

Reply:
Ah ok

Reply:
oh, oops, hehe, I don't but I'm asking at work if anyone does.

Reply:
I can try it out tonight, so if you confirm that it works let me know but otherwise I'll check then (8-10 hours from now).